text/0011-fluent-in-prefs-design-review.md

```diff
@@ -0,0 +1,61 @@
+
+The XBL replacement patch author ([myk](https://mozillians.org/en-US/u/myk/)) and reviewer ([jaws](https://mozillians.org/en-US/u/jaws/)) were both members of the review team, and they noted that those changes should land soon and thus shouldn't create schedule risk.
+
+Another member of the review team noted that another team is rewriting the "delete cookies" dialogs, and the engineering team should coordinate with that team.
+
+## Migration
+
+The teams discussed tools for migrating from DTD/properties files to FTL files. They also discussed the possibility of migrating strings from one FTL file to another, as the difficulty of migrating strings between DTD/properties files is a common frustration with the current system.
+
+The engineering team has developed tools to migrate strings from DTD/properties to FTL files, but they don't yet have a way to migrate them from one FTL file to another. That doesn't regress functionality, but the review team still recommended that the engineering team consider tackling that problem in the future.
+
+The teams also discussed merging the various preferences DTD/properties files into a single FTL file, since their current ontology is out-of-date (because pref categories have changed over time, and the files haven't changed with them), and since there isn't a reason to load only a subset of strings (especially given the new search feature, which needs to search across all of them).
+
+There was also some discussion about the conflict between engineers, who bias toward coalescing "identical" English strings used in different contexts (such as "open") because DRY; and localizers, who separate such strings because they aren't actually always identical in all locales.
+
+Coalescing is considered harmful, and a concern was raised that merging preferences strings into a single FTL file might encourage engineers to do it even more. The teams concluded that this concern can be addressed at the policy/review level.
+
+## String Changes
+
+The review team probed about Fluent support for string changes, and the engineering team explained that support is as good as the current system, which has two ways to modify a string, and work is planned to make it even better by adding a third way.
+
+Specifically, under both the current system and the current version of Fluent, a developer can make a minor change to a string (f.e. a typo fix) that shouldn't affect its translations by preserving the string's key, and they can make a major change to a string that should affect its translations by changing the string's key.
+
+In the future, Fluent will support versioning of a string key via a tag, such that developers can also make changes that affect localizations without having to change the string's key. (There was some discussion about which kinds of changes would fit into this bucket.)
```

Comment:
I believe the Fluent mechanism for this is a special comment, but I don't remember the term; perhaps worth putting it here so we remember in the future.

Reply:
We call it `semantic comments`. The idea is that you could do:

```
key = Value
```

and then later:

```
# @version: 2
key = New Value
```

Runtime would just pick the `key` and ignore the comments, while tooling could notify the localizer that his localization was implicitly marked as `version: 1` and the en-US has `version: 2` so he may want to update the translation.

This is just a concept at this point and I hope to write down the plan for this in Q1.